File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
import math as mt
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack  =[]
        falg =  False
        for i in tokens :
            try :
                int(i)
                flag = True
            except :
                flag = False
            if flag:
                stack.append(int(i))
            else:
                x = int(stack.pop())
                y = int(stack.pop())
                if i =="*":
                    stack.append(y*x)
                elif i =="+":
                    stack.append(y+x)
                elif i == "-":
                    stack.append(y-x)
                else:
                    stack.append((y/x))
        #print(stack)
        return int(stack[-1])

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.cpp
#include <string>
#include <cctype>
#include <stdlib.h>
class Solution {
public:
    int evalRPN(vector<string>& t) {
        vector<int> stack;
        vector<int> num = {1,2,3,4,5,6,7,8,9};
        bool flag = false;
        for(auto &i : t){

            if(i == "+" || i == "-" || i == "/" || i == "*"){
                int x =  stack.back();
                stack.pop_back();
                int y =  stack.back();
                stack.pop_back();
                if(i == "+") stack.push_back(x+y);
                else if(i=="*") stack.push_back(y*x);
                else if(i == "-") stack.push_back(y-x);
                else stack.push_back(y/x);
            }
            else{
           stack.push_back(stoi(i));

            }
        }
        return stack.back();
    }
};